### ISSUE TITLE
Problem(s) with batch code export from arrangement

### ISSUE THREAD

**Issue by codemanyak** [fesch/Structorizer.Desktop#836]
On exporting an arrangement to Pascal in batch mode according to the syntax introduced with #696, e.g.
```bash
structorizer.sh -x Pascal ~/tests/SORTING_TEST.arrz!maxHeapify(3)!quickSort(3)
```
the resulting file is inconsistent, as it will contain as many UNIT definitions as there are valid signatures of procedures or functions not mutually calling each other in the arrangement specification. Only the first of them will contain all subroutines directly or indirectly called by any of them, the latter won't (see attachment).
[test4.zip](https://github.com/fesch/Structorizer.Desktop/files/4334187/test4.zip)

So, on decomposing the resulting Pascal file afterwards, the user would have to copy (or even move) the needed routine definitions from one UNIT to the other.
For other export languages it may not be a problem to share the subroutines in the same file, but in Pascal we would either have to create a common UNIT (if only routines are specified) or to produce several minimum complete modules (UNITs/PROGRAMs) as cuttable sections in the file.

For object-oriented export languages (like C++ or Java) trouble may also arise if a main program and some not automatically involved routines or several main programs in the same arrangement are elected. 

These problems are also relevant for issue #828.

**Comment by codemanyak** [comment]
The conceptional difficulties and contradictions to solve this issue are explained in https://github.com/fesch/Structorizer.Desktop/issues/828#issuecomment-599926355. Here I discuss the additional implications of batch export:

For the batch export of arrangements, a syntax had been introduced to specify entry points of arrangement (see command line example in the issue description https://github.com/fesch/Structorizer.Desktop/issues/836#issue-581656981). This leads to additional implications.
Recently, there were two cases if no entry points were specified:
1. There is exactly a single main diagram in the arrangement, then it will be used as entry point. Just this diagram together with all required diagrams in topologically sorted order would be exported, Not required diagrams would not appear in the code file.
2. There is no main diagram or more then one main diagrams in the arrangement, then nothing was exported because of ambiguity.

Now we should change this behaviour according to the case analysis given in https://github.com/fesch/Structorizer.Desktop/issues/828#issuecomment-599926355.

With specified entry points, however, the originally intended (but inconsistently implemented) behaviour was to export all the entry points (separated by scissor lines) have their code preceded by their topologically sorted required subroutines. These are the additional aspects of the explicit entry point specification:
- Specified entry points might not be roots of the dependency trees identified in the arrangement but some internal nodes or even leaves.
- Diagrams not required by (i.e. not being member of) any of the (sub-)trees rooted by the specified entry points would not be exported. This behaviiour should be adhered to.
- If some of the specified entry points are also members of the tree of some other entry point, then they would only occur within the subroutine sequence. In the task of aligning the behaviour with #828, they would indeed have to go to the common library and have to be marked as public. Instead of some wrapper, just a comment where to find the routine ought to be inserted between scissor lines.

**Comment by codemanyak** [comment]
Supposed to be fixed with version 3.30-07
